src/cbor.c

```diff
@@ -185,10 +185,131 @@ cn_cbor * cn_cbor_null_create(CBOR_CONTEXT_COMMA cn_cbor_errback * errp)
 }
 
 
-unsigned char RgbDontUse4[8 * 1024];
+#ifndef CBOR_NO_FLOAT
+static size_t _size_double(double val)
+{
+	size_t size = 9;
+	float floatVal = val;
+	if (floatVal == val) {
+		bool half = false;
+		union {
+			float f;
+			uint32_t u;
+		} u32;
+		u32.f = floatVal;
+		/* Check for half */
+		if ((u32.u & 0x1FFF) == 0) {
+			int exp = (u32.u >> 23) & 0xff;
+			int mant = u32.u & 0x7fffff;
+			half = (mant == 0 && (exp == 0 || exp == 255)) || 
+			          (exp >= 113 && exp <= 142)  || 
+			          ((exp >= 103 && exp < 113) && !(mant & ((1 << (126 - exp)) - 1)));
+		}
+		if(half) {
+			size = 3;
+		} else {
+			size = 5;
+		}
+	} else if (val != val) {
+		size = 3;
+	}
+
+	return size;
+}
+#endif /* CBOR_NO_FLOAT */
+
+static size_t _size_positive(uint64_t val)
+{
+	size_t size;
+
+	if (val < 24) {
+		size = 1;
+	} else if (val < 256) {
+		size = 2;
+	} else if (val < 65536) {
+		size = 3;
+	} else if (val < 0x100000000) {
+		size = 5;
+	} else {
+		size = 9;
+	}
+
+	return size;
+}
 
 size_t cn_cbor_encode_size(cn_cbor * object)
 {
-	ssize_t size = cn_cbor_encoder_write(RgbDontUse4, 0, sizeof(RgbDontUse4), object);
-	return size >= 0 ? size : 0;
+	size_t size = 0;
```

Comment:
I had always planned to do this by allowing for NULL to be passed in to cn_cbor_encoder_write as the desitination and have it return the size.  Would that make more sense?

Reply:
If that function was part of this project, I probably would have done it that way. This works within the limitations of the current cn_cbor code. If/when that changes to support passing NULL, I would be happy to have this code replaced by that.

Reply:
Note that this project uses a version of cn_cbor which is in my repository and not the base version. 

  project_cn-cbor
  GIT_REPOSITORY https://github.com/jimsch/cn-cbor
  GIT_TAG complete